code/modules/roguetown/roguejobs/artificer/gadgets.dm

```diff
@@ -0,0 +1,329 @@
+	else if(!is_loaded && in_use && grappled_turf && tile_effect)	//Reset option.
+		user.visible_message("[user] unhooks from the tile.")
+		reset_tile()
+		reset_target()
+
+//Resets the tile effect and the grappled turf. Generally called with reset_target()
+/obj/item/grapplinghook/proc/reset_tile(silent = FALSE)
+	if(tile_effect && grappled_turf)
+		grappled_turf.cut_overlay(tile_effect)
+		qdel(tile_effect)
+		grappled_turf = null
+	if(!silent)	//Silent is used during a successful reel because it has its own distinct sounds
+		playsound(src, 'sound/foley/trap.ogg', 100, FALSE , 5)
+	is_loaded = FALSE
+	update_icon()
+
+//Resets the target effect overlay and the attached atom. Generally called with reset_tile()
+/obj/item/grapplinghook/proc/reset_target()
+	if(attached && target_effect)
+		attached.cut_overlay(target_effect)
+		qdel(target_effect)
+		attached = null
+	in_use = FALSE
+	update_icon()
+
+/obj/item/grapplinghook/proc/check_path(turf/Tu, turf/Tt)
+	var/dist = get_dist(Tt, Tu)
+	var/last_dir
+	var/turf/last_step = get_step_multiz(Tu, UP)
+	var/success = FALSE
+	for(var/i = 0, i <= dist, i++)
+		last_dir = get_dir(last_step, Tt)
+		var/turf/Tstep = get_step(last_step, last_dir)
+		if(!Tstep.density)
+			success = TRUE
+			var/list/cont = Tstep.GetAllContents(/obj/structure/roguewindow)
+			for(var/obj/structure/roguewindow/W in cont)
+				if(W.climbable && !W.opacity)	//It's climable and can be seen through
+					success = TRUE
+					LAZYADD(obj_to_destroy, W)
+					continue
+				else if(!W.climbable)
+					success = FALSE
+					return success
+		else
+			success = FALSE
+			return success
+		last_step = Tstep
+	return success
+
+
+
+//Successful reel, complete reset.
+/obj/item/grapplinghook/proc/reel()
+	if(attached && in_use && grappled_turf)
+		if(do_teleport(attached, grappled_turf))
+			playsound(attached, 'sound/misc/grapple_reel.ogg', 100, FALSE)
+			playsound(grappled_turf, 'sound/misc/grapple_reel.ogg', 100, FALSE)
+			destroy_eligible_objects()
+			reset_tile(silent = TRUE)
+			reset_target()
+			unload(failure = TRUE)
+
+/obj/item/grapplinghook/proc/destroy_eligible_objects()
+	if(length(obj_to_destroy))
+		for(var/obj/O in obj_to_destroy)
+			if(istype(O,/obj/structure/roguewindow))
+				var/obj/structure/roguewindow/W = O
+				if(!W.climbable)
+					O.obj_integrity = 1	//Keeps it from being destroyed
+					O.obj_break()
+		LAZYCLEARLIST(obj_to_destroy)
+
+/obj/item/grapplinghook/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
+	if(istype(user.used_intent, /datum/intent/grapple))	//First step, grappling onto a tile. Spawns an indicator on it.
+		if(is_loaded && istype(target, /turf/))
+			var/turf/T = target
+			if(!istransparentturf(T) && T.z > user.z) //We are shooting at a floor turf above
+				var/reason
+				if(max_range >= get_dist(user, T) && !T.density)
+					if(check_path(get_turf(user), T))	//We check for opaque turfs or non-climbable windows in the way via a simple pathfind.
```

Comment:
Hey! I made this for Azure Peak originally. Just giving a fair warning since your playerbase seems more gameplay-oriented -- this does not respect gates, tables, bars, shutters and other such structures people aren't meant to get past conventionally. If your map has spots where you can look up and grapple through such, it might end up abused. AP's map had very little such spots, so we got by without added exceptions.

Reply:
> Hey! I made this for Azure Peak originally. Just giving a fair warning since your playerbase seems more gameplay-oriented -- this does not respect gates, tables, bars, shutters and other such structures people aren't meant to get past conventionally. If your map has spots where you can look up and grapple through such, it might end up abused. AP's map had very few spots like that, so we got by without added exceptions.

It doesn't respect gates? Like the large gate people can close? The table is fine- bars I'll probably need to tweak- Shutters?

Thank you for your work on this by the way, it's a rather neat tool.

Reply:
It ONLY checks obj/structures that are windows and turfs. So yeah, if you can see through a gate the grappler could hook through it from below. It's very map specific, so it might not be an issue on RW either.